src/qutip_qip/operations/gateclass.py

```diff
@@ -153,17 +158,45 @@ def __init__(
         classical_controls=None,
         classical_control_value: Optional[int] = None,
         arg_label=None,
+        style=None,
         **kwargs,
     ):
         """
         Create a gate with specified parameters.
         """
-
         self.name = name if name is not None else self.__class__.__name__
         self.targets = None
         self.controls = None
         self.classical_controls = None
 
+        if style is not None:
+            self.text = style.get("text", self.name)
+            self.color = style.get(
+                "color",
+                (
+                    default_theme[self.name]
+                    if self.name in default_theme
+                    else "#000000"
+                ),
+            )
+            self.fontsize = style.get("fontsize", 10)
+            self.fontcolor = style.get("fontcolor", "#FFFFFF")
```

Comment:
It would be nice to move this part to the plotting code, not in the gate files. The default input of style could just be `None`, and all the default settings will be added when plotting. 

This avoids importing the `default_theme` and also avoids having too many attributes in the gate class.